subsys/mpsl/mpsl_init.c

```diff
@@ -181,6 +194,27 @@ static int mpsl_signal_thread_init(const struct device *dev)
 	return 0;
 }
 
-SYS_INIT(mpsl_lib_init, PRE_KERNEL_1, CONFIG_KERNEL_INIT_PRIORITY_DEFAULT);
 SYS_INIT(mpsl_signal_thread_init, POST_KERNEL,
 	 CONFIG_KERNEL_INIT_PRIORITY_DEFAULT);
+#else
```

Comment:
I prefer to not duplicate the initialization code

Reply:
I removed it. The amount of conditional compilations has increased a little